lib/drivers/github.js

```diff
@@ -7,49 +8,109 @@ var util = require('util');
 var base64 = require('../utils/base64');
 var Driver = require('./driver');
 
-function GitHubDriver() {
-    Driver.apply(this, arguments);
+var Blob = require('../models/blob');
+var Branch = require('../models/branch');
+var WorkingState = require('../models/workingState');
+
+// Options for the GitHub Driver
+var GitHubOptions = immutable.Record({
+    // String: ID of the GitHub repository (ex: me/myrepo")
+    repository: null,
+
+    // API endpoint
+    host: 'https://api.github.com',
+
+    // Endpoint for RAW content
+    rawhost: 'https://raw.githubusercontent.com',
+
+    // Authentication for the API
+    username: null,
+    token: null,
+
+    // Include auth token in raw url?
+    includeTokenInRaw: false
+});
+
+
+
+function GitHubDriver(options) {
+    Driver.call(this);
+
+    this.options = new GitHubOptions(options);
+
 }
 util.inherits(GitHubDriver, Driver);
 
 // ---- Implemented methods ----
 
-GitHubDriver.prototype.fetchBlob = function(sha) {};
+// Fetch a blob by its sha
+// @param {sha}
+// @return {Blob}
+GitHubDriver.prototype.fetchBlob = function(sha) {
+    return this.get('git/blobs/' + sha)
+    .then(function(r) {
+        return Blob.createFromBase64(r.content);
+    });
+};
 
-GitHubDriver.prototype.fetchTree = function(branch) {};
+// Fetch a tree from a reference
```

Comment:
Let's choose if we put method doc in driver.js or in our implementation, but let's not do both as before.